### PR TITLE
fix(connector-corda): container image kotlin compilation fails in model

### LINKED ISSUE #584: chore(deps): upgrade to openapi generator 2.2.6
There's been some flickering with the generated files recently because we slipped up with one of the PRs (I think) where the 5.0.0 generator was used by accident.
Would've been better to keep with the exact version from before, BUT that said version is actually a beta one so there is a case to be made for pulling the plug on an upgrade which isn't really a disruptive change, but it will generate a large diff for sure.

Edit: 2.2.6 is the latest version of this tool.

### LINKED ISSUE #763: fix(connector-corda): container image kotlin compilation fails in model
**Describe the bug**

The corda connector jvm backend's container image build is failing on DockerHub due to a compilation error in the generated model code.
This is likely related to the recent downgrade to the standard version of the openapi generator (from 5.0.0 to 5.0.0-beta2).

**To Reproduce**

Build the container image on DockerHub or locally and observe it failing.

**Expected behavior**

Build does not fail.

**Logs/Stack traces**

```sh
Step 5/24 : RUN gradle build
---> Running in f1dbb3aad89c
Welcome to Gradle 6.8!
Here are the highlights of this release:
- Faster Kotlin DSL script compilation
- Vendor selection for Java toolchains
- Convenient execution of tasks in composite builds
- Consistent dependency resolution
For more details see https://docs.gradle.org/6.8/release-notes.html
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :compileKotlin
[91me: /kotlin-spring/src/main/kotlin/org/hyperledger/cactus/plugin/ledger/connector/corda/server/model/JarFile.kt: (29, 5): This type has a constructor, and thus must be initialized here
[0m
> Task :compileKotlin
FAILED
[91m
[0m
[91mFAILURE: [0m
[91mBuild failed with an exception.[0m
[91m
[0m
[91m
[0m
[91m* What went wrong:[0m
[91m
[0m
[91mExecution failed for task ':compileKotlin'.
[0m
[91m> Compilation error. See log for more details[0m
[91m
[0m
[91m
[0m
[91m* Try:
[0m
[91mRun with --stacktrace[0m
[91m option to get the stack trace. Run with --info or [0m
[91m--debug option to get more log output. Run with [0m
[91m--scan to get full insights.[0m
[91m
[0m
[91m
* Get more help at [0m
[91mhttps://help.gradle.org
[0m
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings
[91m
[0m
[91mBUILD FAILED in 3m 41s
[0m
1 actionable task: 1 executed
Removing intermediate container f1dbb3aad89c
The command '/bin/sh -c gradle build' returned a non-zero code: 1
```

**Screenshots**

`N/A`

**Cloud provider or hardware configuration:**

`N/A`

**Operating system name, version, build:**

`N/A`

**Hyperledger Cactus release version or commit (git rev-parse --short HEAD):**

`0.4.0`

**Hyperledger Cactus Plugins/Connectors Used**

Corda

**Additional context**

There's another issue open with globally upgrading the openapi generator to 5.0.0 so that would likely solve this bug as well implicitly.

cc: @takeutak @sfuji822 @hartm @jonathan-m-hamilton @AzaharaC @jordigiam @kikoncuo @jagpreetsinghsasan

### PR DESCRIPTION
**Pull request by travis-payne**
Depends on #993  

This change fixes the compilation error kotlin was throwing when building the docker image.

BREAKING CHANGE: 🧨 We're now using the latest OpenAPI Generator CLI version of 2.2.6, so quite a few autogenerated files have been changed.

- Enums generated by OpenAPI are now lowercase.
- API Clients now need to instantiate a Configuration object and use that as a constructor param.
- Now using Integer as opposed to Number in the OpenAPI config to prevent an error when mapping to BigDecimal

Resolve #763
Resolve #584

### CODE REVIEW

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <version>0.3.0</version>
     <properties>
         <kotlin.version>1.3.30</kotlin.version>
-	    <kotlinx-coroutines.version>1.2.0</kotlinx-coroutines.version>
+        <kotlinx-coroutines.version>1.2.0</kotlinx-coroutines.version>
```
[the remark this answers is not in the document]
Comment (travis-payne) [reply packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/pom.xml:10]: Confirmed the updated version of OpenAPI has caused this change.

File: packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/src/main/kotlin/org/hyperledger/cactus/plugin/ledger/connector/corda/server/api/ApiPluginLedgerConnectorCordaService.kt [hunk 2]
```diff
@@ -10,15 +10,16 @@ import org.hyperledger.cactus.plugin.ledger.connector.corda.server.model.InvokeC
 import org.hyperledger.cactus.plugin.ledger.connector.corda.server.model.ListFlowsV1Request
 import org.hyperledger.cactus.plugin.ledger.connector.corda.server.model.ListFlowsV1Response
 import org.hyperledger.cactus.plugin.ledger.connector.corda.server.model.NodeInfo
+
 interface ApiPluginLedgerConnectorCordaService {
 
-	fun deployContractJarsV1(deployContractJarsV1Request: DeployContractJarsV1Request?): DeployContractJarsSuccessV1Response
+    fun deployContractJarsV1(deployContractJarsV1Request: DeployContractJarsV1Request?): DeployContractJarsSuccessV1Response
```
[the remark this answers is not in the document]
Comment (travis-payne) [reply packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/src/main/kotlin/org/hyperledger/cactus/plugin/ledger/connector/corda/server/api/ApiPluginLedgerConnectorCordaService.kt:16]: Confirmed the updated version of OpenAPI has caused this change.

File: packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/settings.gradle [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-		gradlePluginPortal()
-	}
-	resolutionStrategy {
-		eachPlugin {
-			if (requested.id.id == "org.springframework.boot") {
-				useModule("org.springframework.boot:spring-boot-gradle-plugin:${requested.version}")
-			}
-		}
-	}
+    repositories {
```
[the remark this answers is not in the document]
Comment (travis-payne) [reply packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/settings.gradle:2]: Confirmed the updated version of OpenAPI has caused this change.

**Comment by petermetz** [comment]
@travis-payne FYI these errors are caused by the imports with relative paths that I commented on maybe once, but there could've been a few more that I missed, I'm not sure:
https://github.com/hyperledger/cactus/pull/879/checks?check_run_id=2458629533#step:5:17198

**Review by petermetz — APPROVED** [review]

**Review by takeutak — APPROVED** [review]

**Comment by petermetz** [comment]
@travis-payne 0.5.0 is out now, freeze is over, please rebase + squash this branch and then we can finally merge it in.

**Pull request merged**